rerun_py/rerun_sdk/rerun/archetypes/boxes2d_ext.py

```diff
@@ -57,26 +87,70 @@ def __init__(
         instance_keys:
             Unique identifiers for each individual boxes in the batch.
         """
-
-        if sizes is not None:
+        if array is not None:
             if half_sizes is not None:
-                _send_warning("Cannot specify both `sizes` and `half_sizes` at the same time.", 1)
-
-            sizes = np.asarray(sizes, dtype=np.float32)
-            half_sizes = sizes / 2.0
-
-        if mins is not None:
+                _send_warning("Cannot specify both `array` and `half_sizes` at the same time.", 1)
+            if sizes is not None:
+                _send_warning("Cannot specify both `array` and `sizes` at the same time.", 1)
+            if mins is not None:
+                _send_warning("Cannot specify both `array` and `mins` at the same time.", 1)
             if centers is not None:
-                _send_warning("Cannot specify both `mins` and `centers` at the same time.", 1)
-
-            # already converted `sizes` to `half_sizes`
-            if half_sizes is None:
-                _send_warning("Cannot specify `mins` without `sizes` or `half_sizes`.", 1)
-                half_sizes = np.asarray([1, 1], dtype=np.float32)
-
-            mins = np.asarray(mins, dtype=np.float32)
-            half_sizes = np.asarray(half_sizes, dtype=np.float32)
-            centers = mins + half_sizes
+                _send_warning("Cannot specify both `array` and `centers` at the same time.", 1)
+
+            if np.any(array):
```

Comment:
np.any is documented as 

> Test whether any array element along a given axis evaluates to True.

Don't get it, what does this do here then? Why don't you just always `np.asarray` the incoming array, I think that's what we do everywhere else?